### PR TITLE
PSG-6183: Add a pangolin_data_version column to the end of the pangolin results CSV file

### PR DESCRIPTION
**Pull request by mduk**
```
root@55dc2a1d7f55:/app# nextflow run ./workflows/pangolin.nf --run 61c06b0a-e5e8-4dbf-8bb0-729cce46a223 --config-path /app/local_test/fasta/ --output_path /app/output/pangolin/
N E X T F L O W  ~  version 23.10.1
Launching `./workflows/pangolin.nf` [ridiculous_hoover] DSL2 - revision: 0bd3f5b47c
executor >  local (1)
[fe/cd141a] process > PANGOLIN_PIPELINE (1) [100%] 1 of 1 ✔
[[SAMPLE_ID:4a32dffd-584c-4f2a-8b17-e1e27b630f66, PRIMER:ARTIC_V3], /app/local_test/fasta/4a32dffd-584c-4f2a-8b17-e1e27b630f66.fasta]
Using reference data from: /app/reference/pangolin-data/branch/prerelease/1.23.1/00001/f1b0a3799ebf31edae945d8fab2bd844ecc8952c

root@55dc2a1d7f55:/app# cat work/fe/cd141a8a3dede87064f36da1cdf1ef/pangolin_output/4a32dffd-584c-4f2a-8b17-e1e27b630f66_lineage_report.csv
taxon,lineage,conflict,ambiguity_score,scorpio_call,scorpio_support,scorpio_conflict,scorpio_notes,version,pangolin_version,scorpio_version,constellation_version,is_designated,qc_status,qc_notes,note,pangolin_data_version
4a32dffd-584c-4f2a-8b17-e1e27b630f66,B.1.617.2,,,Delta (B.1.617.2-like),0.38,0.0,scorpio call: Alt alleles 5; Ref alleles 0; Amb alleles 8; Oth alleles 0,SCORPIO_v0.1.12,4.3,0.3.17,v0.1.12,False,fail,Ambiguous_content:0.31,scorpio called lineage B.1.617.2,1.21
```

### CODE REVIEW

**Review by morty — APPROVED** [review]

File: integration_tests/expected_data/sars_cov_2/illumina_unknown/results.csv [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
+0c0b5622-d63a-42cc-a804-13d71e1a5306,Completed,53046,ARTIC_V4-1,550,49,0,0.4757281553398058,unknown,1.11,98.89,19359,501636,True,B.1.369,0.0,,,,,,PUSHER-v1.26,4.3,0.3.17,v0.1.12,False,pass,Ambiguous_content:0.03,Usher placements: B.1.369(1/1),1.21
+39330daf-8dc6-4faa-a2d0-a2c1c909ab3b,Completed,63706,ARTIC_V4-1,471,47,0,0.4563106796116505,unknown,1.88,98.12,19372,494576,True,B.1,0.0,,,,,,PUSHER-v1.26,4.3,0.3.17,v0.1.12,False,pass,Ambiguous_content:0.03,Usher placements: B.1(2/2),1.21
+5695c67c-b7ac-4fc2-96f0-f1ffc68c03fb,Completed,9038,Midnight-ONT_V3,8439,29,0,1.0,unknown,0.79,99.17,22147,459268,True,BA.1,0.0,,Omicron (BA.1-like),0.92,0.0,scorpio call: Alt alleles 53; Ref alleles 0; Amb alleles 2; Oth alleles 3,PUSHER-v1.26,4.3,0.3.17,v0.1.12,False,pass,Ambiguous_content:0.02,Usher placements: BA.1(1/1),1.21
+70fd6fd3-6eed-48a4-8e5d-d59ecb1dbf32,Completed,1990720,NEB-VarSkip_v2b,4640,1,0,0.0133333333333333,unknown,0.0,0.0,0,0,False,Unassigned,,,,,,,PUSHER-v1.26,4.3,0.3.17,v0.1.12,False,fail,failed to map,,1.21
```
Comment (cramshaw) [inline integration_tests/expected_data/sars_cov_2/illumina_unknown/results.csv:5]: it should be 1.21 (as we use ref data 1.26 in the CI)

File: app/modules/pangolin.nf [hunk 2]
```diff
@@ -33,5 +33,13 @@ process PANGOLIN_PIPELINE {
       --outfile ${output_filename} \
       --datadir ${pangolin_data_dir} \
       --use-old-datadir
+
+
+    # Add pangolin_data_version column to the end of the result CSV file
+    declare -r pangolin_data_version=\$(pangolin --pangolin-data-version | cut -d ' ' -f 2)
```
Comment (cramshaw) [inline app/modules/pangolin.nf:39]: you'll need to pass --datadir here as well (like above)

File: integration_tests/expected_data/sars_cov_2/illumina_unknown/results.csv (hunk 1 above)
Comment (mduk) [reply integration_tests/expected_data/sars_cov_2/illumina_unknown/results.csv]: Changed to 1.26

File: app/modules/pangolin.nf (hunk 2 above)
Comment (mduk) [reply app/modules/pangolin.nf]: Added --datadir parameter.

**Pull request merged**